FlattenBatches() doesn't preserve batch validation options during Copy()

### ACH Version

1.55.3

### What were you trying to do?



  When FlattenBatches() consolidates batches, it creates new batches via Copy() but doesn't preserve the batch-level validation options. This causes validation to fail when using options like
  AllowEmptyIndividualName.

  Expected Behavior

  Batch validation options should be preserved when batches are copied during the flatten operation.

  Current Behavior

  1. Original batches have validation options set via batch.SetValidation()
  2. FlattenBatches() calls Copy() which creates new batches without validation options
  3. New batches fail validation when batch.Create() is called
  4. Errors are silently ignored at line 95 in file_flattener.go
  5. Result: no batches are added to the flattened file


### What did you expect to see?

Validation options propagated and new batches created in flattened file

### What did you see?

panic: batch #1 (WEB) FieldError IndividualName  is a mandatory field and has a default value, did you use the constructor? [recovered, repanicked]


### How can we reproduce the problem?

```
// Create batch with validation option
  batch, _ := ach.NewBatch(batchHeader)
  batch.SetValidation(&ach.ValidateOpts{
      AllowEmptyIndividualName: true,
  })

  // Add entry with spaces for individual name
  entryDetail := ach.NewEntryDetail()
  entryDetail.IndividualName = "                      " // 22 spaces
  batch.AddEntry(entryDetail)
  batch.Create() // ✅ Works

  // Add to file
  file := ach.NewFile()
  file.AddBatch(batch)
  file.Create() // ✅ Works

  // Flatten fails because validation options not preserved
  flattenedFile, err := file.FlattenBatches() // ❌ Fails
  // Error: "must have []*Batches or []*IATBatches to be built
```

--- a/file_flattener.go
+++ b/file_flattener.go
@@ -92,7 +92,9 @@ func Flatten(originalFile *File) (*File, error) {
 	sort.Slice(allBatches, func(i int, j int) bool { return allBatches[i].GetBatchNumber() < allBatches[j].GetBatchNumber() })
 
 	for i := range allBatches {
-		allBatches[i].AddToFile(newFile)
+		if err := allBatches[i].AddToFile(newFile); err != nil {
+			return nil, err
+		}
 	}
 
 	if err := newFile.Create(); err != nil {
@@ -208,6 +210,26 @@ func (m mergeableBatcher) AddToFile(file *File) error {
 		return m.batcher.GetEntries()[i].TraceNumber < m.batcher.GetEntries()[j].TraceNumber
 	})
 
+	// Inherit validation options from file if batch doesn't have them
+	if file.GetValidation() != nil {
+		// Use a local interface to access SetValidation
+		type validatable interface {
+			SetValidation(*ValidateOpts)
+		}
+
+		if v, ok := m.batcher.(validatable); ok {
+			v.SetValidation(file.GetValidation())
+
+			// Also set validation on all entries
+			for _, entry := range m.batcher.GetEntries() {
+				entry.SetValidation(file.GetValidation())
+			}
+			for _, entry := range m.batcher.GetADVEntries() {
+				entry.SetValidation(file.GetValidation())
+			}
+		}
+	}
+
 	err := m.batcher.Create()
 	if err != nil {
 		return askForBugReports(fmt.Errorf("mergeableBatcher - AddToFile: %v", err))
@@ -275,6 +297,16 @@ func (m mergeableIATBatch) AddToFile(file *File) error {
 	sort.Slice(m.iatBatch.Entries, func(i, j int) bool {
 		return m.iatBatch.Entries[i].TraceNumber < m.iatBatch.Entries[j].TraceNumber
 	})
+
+	// Inherit validation options from file
+	if file.GetValidation() != nil {
+		m.iatBatch.SetValidation(file.GetValidation())
+
+		// Also set validation on all entries
+		for _, entry := range m.iatBatch.Entries {
+			entry.SetValidation(file.GetValidation())
+		}
+	}
 
 	err := m.iatBatch.Create()
 	if err != nil {

--- a/file_flattener_test.go
+++ b/file_flattener_test.go
@@ -221,3 +221,124 @@ func TestFlattenFile_ValidateOpts(t *testing.T) {
 	require.NotNil(t, opts)
 	require.True(t, opts.SkipAll)
 }
+
+func TestFlattenFile_PropagatesValidationToBatches(t *testing.T) {
+	// Create a file with multiple WEB batches that have entries with empty IndividualName
+	// This tests that AllowEmptyIndividualName is propagated from file to batches during flatten
+
+	file := NewFile()
+	file.SetHeader(mockFileHeader())
+
+	// Create two batches with the same header (so they'll be flattened together)
+	// but with entries that have empty IndividualName
+	for i := 0; i < 2; i++ {
+		bh := NewBatchHeader()
+		bh.ServiceClassCode = MixedDebitsAndCredits
+		bh.CompanyName = "Test Company"
+		bh.CompanyIdentification = "1234567890"
+		bh.StandardEntryClassCode = WEB
+		bh.CompanyEntryDescription = "Test"
+		bh.ODFIIdentification = "12104288"
+
+		batch, err := NewBatch(bh)
+		require.NoError(t, err)
+
+		// Set validation to allow empty names during batch creation
+		batch.SetValidation(&ValidateOpts{
+			AllowEmptyIndividualName: true,
+		})
+
+		// Add entry with empty IndividualName (only spaces)
+		entry := NewEntryDetail()
+		entry.TransactionCode = CheckingCredit
+		entry.RDFIIdentification = "12345678"
+		entry.CheckDigit = "0"
+		entry.DFIAccountNumber = "123456789"
+		entry.Amount = 100000
+		entry.IndividualName = "                      " // 22 spaces - empty name
+		entry.TraceNumber = fmt.Sprintf("121042880000%03d", i+1)
+		entry.IdentificationNumber = "location1234567"
+
+		batch.AddEntry(entry)
+		require.NoError(t, batch.Create())
+
+		file.AddBatch(batch)
+	}
+
+	require.NoError(t, file.Create())
+
+	// Now set validation on the file and flatten
+	file.SetValidation(&ValidateOpts{
+		AllowEmptyIndividualName: true,
+	})
+
+	// Flatten should succeed because validation is propagated to batches
+	flattenedFile, err := Flatten(file)
+	require.NoError(t, err)
+	require.NotNil(t, flattenedFile)
+
+	// Verify we have one batch (two were merged)
+	require.Equal(t, 1, len(flattenedFile.Batches))
+
+	// Verify the batch has both entries
+	require.Equal(t, 2, len(flattenedFile.Batches[0].GetEntries()))
+
+	// Verify entries still have empty names
+	for _, entry := range flattenedFile.Batches[0].GetEntries() {
+		require.Equal(t, "                      ", entry.IndividualName)
+	}
+
+	// Verify file-level validation was preserved
+	opts := flattenedFile.GetValidation()
+	require.NotNil(t, opts)
+	require.True(t, opts.AllowEmptyIndividualName)
+}
+
+func TestFlattenFile_ValidationPropagationFailsWithoutOpts(t *testing.T) {
+	// This test verifies that without AllowEmptyIndividualName,
+	// flattening batches with empty names will fail
+
+	file := NewFile()
+	file.SetHeader(mockFileHeader())
+
+	// Create batch with empty IndividualName
+	bh := NewBatchHeader()
+	bh.ServiceClassCode = MixedDebitsAndCredits
+	bh.CompanyName = "Test Company"
+	bh.CompanyIdentification = "1234567890"
+	bh.StandardEntryClassCode = WEB
+	bh.CompanyEntryDescription = "Test"
+	bh.ODFIIdentification = "12104288"
+
+	batch, err := NewBatch(bh)
+	require.NoError(t, err)
+
+	// Initially allow empty names to create the batch
+	batch.SetValidation(&ValidateOpts{
+		AllowEmptyIndividualName: true,
+	})
+
+	entry := NewEntryDetail()
+	entry.TransactionCode = CheckingCredit
+	entry.RDFIIdentification = "12345678"
+	entry.CheckDigit = "0"
+	entry.DFIAccountNumber = "123456789"
+	entry.Amount = 100000
+	entry.IndividualName = "                      " // 22 spaces - empty name
+	entry.TraceNumber = "121042880000001"
+	entry.IdentificationNumber = "location1234567"
+
+	batch.AddEntry(entry)
+	require.NoError(t, batch.Create())
+
+	file.AddBatch(batch)
+	require.NoError(t, file.Create())
+
+	// Don't set validation on file (or set it without AllowEmptyIndividualName)
+	file.SetValidation(&ValidateOpts{})
+
+	// Flatten should fail because validation is not allowing empty names
+	_, err = Flatten(file)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "IndividualName")
+}

